### DISCUSSION TITLE
How to add external / third party library .jars to pods?

### DISCUSSION THREAD

**Discussion by polsson12** [JanssenProject/jans#9254] [Q&A] [answered]
In a multi node kubernetes cluster environment with terraform deployment, how do you add external java library jars to the pods and update the corresponding xml-config?  

It's pretty well outlined [here in the docs](https://docs.jans.io/v1.1.2/admin/developer/interception-scripts/#using-java-libraries-in-a-jython-script) how you do it manually if you have access to the machines, i.e., 

> Steps: 
> 1. Add library jars to /opt/jans/jetty/jans-auth/custom/libs/ 
> 3. Edit /opt/jans/jetty/jans-auth/webapps/jans-auth.xml and add the following line replacing the word library-name with the actual name of the library:
> `<Set name="extraClasspath">/opt/jans/jetty/jans-auth/custom/libs/library-name.jar</Set>`
> 4. Restart jans-auth service systemctl restart jans-auth
> 

But how do you achieve the same with in an automated CI/CD deployment pipeline deploying to a multi node kubernetes cluster with terraform?

Thanks!

**Comment by nynymike** [comment]
@misba7  maybe you can share your thoughts here?

**Comment by misba7** [comment, marked as answer]
Hi!

You can find it well-documented [here](https://docs.jans.io/v1.1.5.nightly/admin/kubernetes-ops/customization/#custom-jar-file-for-scripts).

The above link explains how to add custom jars in the context of Janssen Cloud-Native/K8s deployment.

**Reply by polsson12** [reply]
Thank you very much! Seems to be what I'm looking for!

**Reply by misba7** [reply]
You're most welcome!
